test/e2e/traces_service_name_test.go

```diff
@@ -71,73 +67,50 @@ var _ = Describe(suite.ID(), Label(suite.LabelTraces), func() {
-			verifyServiceNameAttr(servicenamebundle.DaemonSetName, servicenamebundle.DaemonSetName)
+			assert.VerifyServiceNameAttr(proxyClient, backendExportURL, servicenamebundle.DaemonSetName, servicenamebundle.DaemonSetName)
 		})
 
 		It("Should set undefined service.name attribute to Job name", func() {
-			verifyServiceNameAttr(servicenamebundle.JobName, servicenamebundle.JobName)
+			assert.VerifyServiceNameAttr(proxyClient, backendExportURL, servicenamebundle.JobName, servicenamebundle.JobName)
 		})
 
 		It("Should set undefined service.name attribute to Pod name", func() {
-			verifyServiceNameAttr(servicenamebundle.PodWithNoLabelsName, servicenamebundle.PodWithNoLabelsName)
+			assert.VerifyServiceNameAttr(proxyClient, backendExportURL, servicenamebundle.PodWithNoLabelsName, servicenamebundle.PodWithNoLabelsName)
 		})
 
 		It("Should enrich service.name attribute when its value is unknown_service", func() {
-			verifyServiceNameAttr(servicenamebundle.PodWithUnknownServiceName, servicenamebundle.PodWithUnknownServiceName)
+			assert.VerifyServiceNameAttr(proxyClient, backendExportURL, servicenamebundle.PodWithUnknownServiceName, servicenamebundle.PodWithUnknownServiceName)
 		})
 
 		It("Should enrich service.name attribute when its value is following the unknown_service:<process.executable.name> pattern", func() {
-			verifyServiceNameAttr(servicenamebundle.PodWithUnknownServicePatternName, servicenamebundle.PodWithUnknownServicePatternName)
+			assert.VerifyServiceNameAttr(proxyClient, backendExportURL, servicenamebundle.PodWithUnknownServicePatternName, servicenamebundle.PodWithUnknownServicePatternName)
 		})
 
 		It("Should NOT enrich service.name attribute when its value is not following the unknown_service:<process.executable.name> pattern", func() {
-			verifyServiceNameAttr(servicenamebundle.PodWithInvalidStartForUnknownServicePatternName, servicenamebundle.AttrWithInvalidStartForUnknownServicePattern)
-			verifyServiceNameAttr(servicenamebundle.PodWithInvalidEndForUnknownServicePatternName, servicenamebundle.AttrWithInvalidEndForUnknownServicePattern)
-			verifyServiceNameAttr(servicenamebundle.PodWithMissingProcessForUnknownServicePatternName, servicenamebundle.AttrWithMissingProcessForUnknownServicePattern)
+			assert.VerifyServiceNameAttr(proxyClient, backendExportURL, servicenamebundle.PodWithInvalidStartForUnknownServicePatternName, servicenamebundle.AttrWithInvalidStartForUnknownServicePattern)
+			assert.VerifyServiceNameAttr(proxyClient, backendExportURL, servicenamebundle.PodWithInvalidEndForUnknownServicePatternName, servicenamebundle.AttrWithInvalidEndForUnknownServicePattern)
+			assert.VerifyServiceNameAttr(proxyClient, backendExportURL, servicenamebundle.PodWithMissingProcessForUnknownServicePatternName, servicenamebundle.AttrWithMissingProcessForUnknownServicePattern)
 		})
 
 		It("Should have no kyma resource attributes", func() {
-			Eventually(func(g Gomega) {
```

Comment:
If this logic is only used in this check, then you don't need to extract into a separate function.